testing/base/source_gen.cpp

```diff
@@ -125,75 +125,76 @@ auto SourceGen::GenAPIFileDenseDecls(int target_lines, DenseDeclParams params)
+  return idents;
 }
 
-auto SourceGen::GetShuffledUniqueIds(int number, int min_length, int max_length,
-                                     bool uniform)
+auto SourceGen::GetShuffledUniqueIdentifiers(int number, int min_length,
+                                             int max_length, bool uniform)
     -> llvm::SmallVector<llvm::StringRef> {
   CARBON_CHECK(min_length >= 4)
       << "Cannot trivially guarantee enough distinct, unique identifiers for "
          "lengths <= 3";
-  llvm::SmallVector<llvm::StringRef> ids =
-      GetUniqueIds(number, min_length, max_length, uniform);
-  std::shuffle(ids.begin(), ids.end(), rng_);
-  return ids;
+  llvm::SmallVector<llvm::StringRef> idents =
+      GetUniqueIdentifiers(number, min_length, max_length, uniform);
+  std::shuffle(idents.begin(), idents.end(), rng_);
+  return idents;
 }
 
-auto SourceGen::GetIds(int number, int min_length, int max_length, bool uniform)
+auto SourceGen::GetIdentifiers(int number, int min_length, int max_length,
+                               bool uniform)
     -> llvm::SmallVector<llvm::StringRef> {
-  llvm::SmallVector<llvm::StringRef> ids =
-      GetIdsImpl(number, min_length, max_length, uniform,
-                 [this](int length, int length_count,
-                        llvm::SmallVectorImpl<llvm::StringRef>& dest) {
-                   auto length_ids = GetSingleLengthIds(length, length_count);
-                   dest.append(length_ids.begin(), length_ids.end());
-                 });
-
-  return ids;
+  llvm::SmallVector<llvm::StringRef> idents = GetIdentifiersImpl(
+      number, min_length, max_length, uniform,
+      [this](int length, int length_count,
+             llvm::SmallVectorImpl<llvm::StringRef>& dest) {
+        auto length_idents = GetSingleLengthIdentifiers(length, length_count);
+        dest.append(length_idents.begin(), length_idents.end());
+      });
+
+  return idents;
 }
 
-auto SourceGen::GetUniqueIds(int number, int min_length, int max_length,
-                             bool uniform)
+auto SourceGen::GetUniqueIdentifiers(int number, int min_length, int max_length,
+                                     bool uniform)
     -> llvm::SmallVector<llvm::StringRef> {
   CARBON_CHECK(min_length >= 4)
       << "Cannot trivially guarantee enough distinct, unique identifiers for "
          "lengths <= 3";
-  llvm::SmallVector<llvm::StringRef> ids =
-      GetIdsImpl(number, min_length, max_length, uniform,
-                 [this](int length, int length_count,
-                        llvm::SmallVectorImpl<llvm::StringRef>& dest) {
-                   AppendUniqueIdentifiers(length, length_count, dest);
-                 });
-
-  return ids;
+  llvm::SmallVector<llvm::StringRef> idents =
+      GetIdentifiersImpl(number, min_length, max_length, uniform,
+                         [this](int length, int length_count,
+                                llvm::SmallVectorImpl<llvm::StringRef>& dest) {
```

Comment:
Why SmallVectorImpl instead of SmallVector? (actually, I see this in a few spots)

Reply:
This is the idiom of passing a small vector reference for appending without encoding a specific small size.